code/game/machinery/turrets.dm

```diff
@@ -288,6 +288,31 @@
 		src.die()
 	return
 
+/obj/machinery/turret/attack_animal(mob/living/simple_animal/M as mob)
+	if(M.melee_damage_upper == 0)
+		return
+	if(!(stat & BROKEN))
+		M.do_attack_animation(src, M)
+		visible_message("<span class='danger'>[M] [M.attacktext] [src]!</span>")
+		M.attack_log += text("\[[time_stamp()]\] <font color='red'>attacked [src.name]</font>")
+		//src.attack_log += text("\[[time_stamp()]\] <font color='orange'>was attacked by [M.name] ([M.ckey])</font>")
+		src.health -= M.melee_damage_upper
+		if (src.health <= 0)
+			src.die()
+	else
+		to_chat(M, "<span class='warning'>That object is useless to you.</span>")
+
+/obj/machinery/turret/attack_alien(mob/living/carbon/alien/humanoid/M as mob)
+	if(!(stat & BROKEN))
+		M.do_attack_animation(src, M)
+		playsound(src, 'sound/weapons/slash.ogg', 25, 1, -1)
+		visible_message("<span class='danger'>[] has slashed at []!</span>", M, src)
```

Comment:
My eyes
`visible_message("<span class='danger'>[M] has slashed at [src]!</span>")` instead, please

Reply:
That wasn't part of the changes, just displacing this code block since it's regular turret stuff, but sure